src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,651 @@
+									break;
+								}
+		
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+										" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}
+						}
+					
+						for (String ej:nosql.getUrlEncErrorInjection()) {
+
+							onlyExactMatch=true;
+							attackType = CRASH_ATTACK;
+							injectedMsg = getNewMsg();
+							valueInjected = ej;
+							setParameter(injectedMsg, param, valueInjected);
+							
+							try {
+								sendAndReceive(injectedMsg, false);
+								
+								if(isBingo(baseMsg, injectedMsg, param, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+		
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+										" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}
+						}
+						
+					
+						for(String[] tvi:nosql.getUrlEncTimedValueInjection()) {
+							attackType = TIMED_ATTACK;
+							injectedMsg = getNewMsg();
+							valueInjected =tvi[0];
+							setParameter(injectedMsg, param, valueInjected);
+							
+							
+							
+							try {
+								now = Instant.now();
+								sendAndReceive(injectedMsg, false);
+								intervalInjectedMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+																
+								if(isTimedInjected(intervalBaseMessage,intervalInjectedMessage, SLEEP_TIME_SHORT)) {
+									
+									// try for a longer time to exclude transmission delays  
+									
+									injectedMsg = getNewMsg();
+									valueInjected =tvi[1];
+									setParameter(injectedMsg, param, valueInjected);
+									
+									now = Instant.now();
+									sendAndReceive(injectedMsg, false);
+									intervalInjectedMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+									
+									if(isTimedInjected(intervalBaseMessage,intervalInjectedMessage, SLEEP_TIME_LONG)) {
+										
+										String extraInfo = Constant.messages.getString(EXTRAINFO_PREFIX + nosql.prefix 
+												+ attackType);
+										String name = getName() + " - " + nosql.getName();
+										
+										bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH, name, getDescription(), null, 
+												param, valueInjected, extraInfo,  getSolution(), injectedMsg);
+										break;
+									}
+								}
+								
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+										" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}						
+						}
+
+					}
+					
+					/**
+					 * TODO is content-type json format?

```

Comment:
Shouldn't this be done much earlier? Is there any point executing the previous set of tests if the original (`getBaseMsg`) CT was JSON?

Reply:
Could do something like:
https://github.com/zaproxy/zap-extensions/blob/e24f58a1094442619ea6a59c456e6dd099ef616d/src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java#L416-L417